api/api_test.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/ausocean/openfish/api/api"
+	"github.com/ausocean/openfish/api/ds_client"
+	"github.com/ausocean/openfish/api/handlers"
+	"github.com/gofiber/fiber/v2"
+)
+
+func TestCreateCaptureSource(t *testing.T) {
+	// Test Setup.
+	ds_client.Init(false)
+	app := fiber.New()
+	RegisterAPIRoutes(app)
+
+	// Send test request.
+	body := strings.NewReader(`{ "name": "Camera 1", "location": "-37.12345678,140.12345678" }`)
+	req := httptest.NewRequest("POST", "/api/v1/capturesources", body)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := app.Test(req, -1)
+	if err != nil {
+		t.Log(err.Error())
+		t.Fail()
+		return
+	}
+
+	// Check response.
+	if resp.StatusCode != fiber.StatusOK {
+		t.Log("expected HTTP code 200, got", resp.StatusCode)
+		t.Fail()
+	}
+}
+
+func TestGetCaptureSources(t *testing.T) {
+	// Test Setup.
+	ds_client.Init(false)
+	app := fiber.New()
+	RegisterAPIRoutes(app)
+
+	// Populate datastore.
+	body := strings.NewReader(`{ "name": "Camera 1", "location": "-37.12345678,140.12345678" }`)
+	req := httptest.NewRequest("POST", "/api/v1/capturesources", body)
+	req.Header.Set("Content-Type", "application/json")
+
+	app.Test(req, -1)
+
+	// Send request
+	req = httptest.NewRequest("GET", "/api/v1/capturesources?limit=1&offset=0", nil)
+	resp, err := app.Test(req)
+	if err != nil {
+		t.Log(err.Error())
+		t.Fail()
+		return
+	}
+
+	// Check response.
+	valid := true
+
+	if resp.StatusCode != fiber.StatusOK {
+		t.Log("expected HTTP code 200, got", resp.StatusCode)
+		valid = false
+	}
+
+	// Unmarshal response.
+	respBody, _ := ioutil.ReadAll(resp.Body)
+	var result api.Result[handlers.CaptureSourceResult]
+	json.Unmarshal(respBody, &result)
+	t.Log("json response: ", string(respBody))
+
+	if len(result.Results) != 1 {
+		t.Log("expected 1 result, got", len(result.Results), "results")
```

Comment:
In each of these cases where you've logged a problem and set valid = false, you can instead use a single call to t.Errorf().